packages/async-flow/test/async-flow-debug.test.js

```diff
@@ -0,0 +1,219 @@
+/* eslint-env node */
+// eslint-disable-next-line import/order
+import { test, getBaggage, annihilate } from './prepare-test-env-ava.js';
+
+import { passStyleOf } from '@endo/pass-style';
+import { makePromiseKit } from '@endo/promise-kit';
+import { eventLoopIteration } from '@agoric/internal/src/testing-utils.js';
+import { isVow } from '@agoric/vow/src/vow-utils.js';
+import { prepareVowTools } from '@agoric/vow';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { prepareTestAsyncFlowTools } from './_utils.js';
+import { consoleRecorder, originalConsole } from './console-recorder-shim.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone';
+ * @import {ExecutionContext} from 'ava';
+ */
+
+/**
+ * @param {ExecutionContext<unknown>} t
+ * @param {any} hOrch
+ * @param {any} reason
+ * @param {any[]} [recording]
+ */
+const rejectionRecordingAssertions = (t, hOrch, reason, recording) => {
+  assert(recording);
+
+  // console.log(
+  //   '@@@ recording was:',
+  //   recording.map(({ method, args }, i) => ({ i, method, args })),
+  // );
+  const cons = (method, ...args) => ({
+    method,
+    args,
+  });
+
+  t.true(recording.every(entry => entry.thisArg === originalConsole));
+
+  // The recording must have at least the entries we have in the expected
+  // pattern below.
+  const recLen17 = Math.max(recording.length, 17);
+
+  // Check the guestMethod stack trace exists.
+  const indexToStackRegex = {
+    2: /\bdoReject\b/,
+    8: /\sIn "reject" method of \(Orchestra\)\s/,
+    [recLen17 - 5]: /\basync-flow-debug\b/,
+    [recLen17 - 2]: /\basyncFlow\b.*\btestRejectionPlay\b/s,
+  };
+
+  const consStack = i => {
+    const re = indexToStackRegex[i];
+    delete indexToStackRegex[i];
+    t.truthy(re, `recorded entry ${i} has a stack regex`);
+    t.is(recording[i].args.length, 1, `recorded entry ${i} has one arg`);
+    t.is(
+      typeof recording[i].args[0],
+      'string',
+      `recorded entry ${i} arg is a string`,
+    );
+    t.regex(recording[i].args[0], re, `recorded entry ${i} maches stack regex`);
+    return { method: 'error' };
+  };
+
+  const consSentAs = i => {
+    t.is(recording[i].args.length, 3, `recorded entry ${i} has three args`);
+    return {
+      method: 'error',
+      args: ['Error#2 ERROR_NOTE:', 'Sent as'],
+    };
+  };
+
+  const error = (...args) => cons('error', ...args);
+  const group = (...args) => cons('group', ...args);
+
+  const expectedRecording = {
+    length: recLen17,
+    0: error('p2 rejection:', '(Error#1)'),
+    1: error('Error#1:', reason.message),
+    2: consStack(2),
+    3: error('Error#1 ERROR_NOTE:', 'from host error', '(Error#2)'),
+    4: error(
+      'Error#1 ERROR_NOTE:',
+      'from flow',
```

Comment:
would it read better as?

```suggestion
      'during execution of flow',
```